move/blackjack/sources/single_player_blackjack.move

```diff
@@ -318,26 +331,30 @@ module blackjack::single_player_blackjack {
 
     /// Function to be called by user who wants to ask for a hit.
     /// @param game: The Game object
-    public fun do_hit(game: &mut Game, current_hand_sum: u8, ctx: &mut TxContext) {
+    public fun do_hit(game: &mut Game, current_player_sum: u8, ctx: &mut TxContext): HitRequest {
         assert!(game.status == IN_PROGRESS, EGameHasFinished);
         assert!(tx_context::sender(ctx) == game.player, EUnauthorizedPlayer);
+        assert!(current_player_sum == game.player_sum, EInvalidSumOfHitRequest);
 
-        event::emit(HitRequested {
-            game_id: object::uid_to_inner(&game.id),
-            current_player_hand_sum: current_hand_sum
-        });
+        HitRequest {
+            id: object::new(ctx),
+            game_id: object::id(game),
+            current_player_sum: game.player_sum,
+        }
     }
 
     /// Function to be called by user who wants to stand.
     /// @param game: The Game object
-    public fun do_stand(game: &mut Game, player_hand_sum: u8, ctx: &mut TxContext) {
+    public fun do_stand(game: &mut Game, current_player_sum: u8, ctx: &mut TxContext): StandRequest {
```

Comment:
Is the current_player_sum argument really necessary here? Is it used as some kind of double-check with the backend to make sure the data is in sync?
(same for above do_hit function)

Reply:
yes yes that's the reason, as a simple double-check